code/controllers/subsystem/monitor.dm

```diff
@@ -0,0 +1,125 @@
+	var/human_in_FOB = 0
+	///The number of time most of humans are in FOB consecutively
+	var/humans_all_in_FOB_counter = 0
+	///TRUE if we detect a state of FOB hugging
+	var/FOB_hugging = FALSE
+	
+
+
+/datum/controller/subsystem/monitor/Initialize(time, zlevel)
+	if(CONFIG_GET(flag/monitor_disallowed))
+		can_fire = 0
+	reschedule()
+	return ..()
+
+/datum/controller/subsystem/monitor/fire(resumed = 0)
+	if(!resumed)
+		check_state() //only check these if we aren't resuming a paused fire
+
+
+///checks if we should refresh the director state, and reschedules if necessary
+/datum/controller/subsystem/monitor/proc/check_state()
+	if(scheduled <= world.time)
+		etablish_state()
+		reschedule()
+
+///decides which world.time we should calculate a new monitor state at.
+/datum/controller/subsystem/monitor/proc/reschedule()
+	scheduled = world.time + FREQUENCY_STATE_CALCULATION
+
+///Start the calculation
+/datum/controller/subsystem/monitor/proc/etablish_state()
+	process_human_positions()
+	current_points = calculate_state_points() / max(GLOB.alive_human_list.len + GLOB.alive_xeno_list.len,20)//having less than 20 players gives bad results
+	FOB_hugging_check()
+	set_state(current_points)
+	//spam_admins() //Only for testing
+
+///Messages admin with the current state
+/datum/controller/subsystem/monitor/proc/spam_admins()
+	message_admins("New state calculated by the monitor, state : [current_state] , exact balance points : [current_points]")
+
+	
+///Calculate the points supposedly representating of the situation	
+/datum/controller/subsystem/monitor/proc/calculate_state_points()
+	var/actualPoints = 0
+	var/datum/job/xeno_job = SSjob.GetJobType(/datum/job/xenomorph)
+	actualPoints += GLOB.monitor_statistics.Ancient_T2 * ANCIENT_T2_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Ancient_T3 * ANCIENT_T3_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Elder_T2 * ELDER_T2_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Elder_T3 * ELDER_T3_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Ancient_Queen * ANCIENT_QUEEN_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Elder_Queen * ELDER_QUEEN_WEIGHT
+	actualPoints += human_on_ground * HUMAN_LIFE_ON_GROUND_WEIGHT
+	actualPoints += (GLOB.alive_human_list.len - human_on_ground) * HUMAN_LIFE_ON_SHIP_WEIGHT
+	actualPoints += GLOB.alive_xeno_list.len * XENOS_LIFE_WEIGHT
+	actualPoints += (xeno_job.total_positions - xeno_job.current_positions) * BURROWED_LARVA_WEIGHT
+	actualPoints += GLOB.monitor_statistics.Miniguns_in_use.len * MINIGUN_PRICE * REQ_POINTS_WEIGHT
+	actualPoints += GLOB.monitor_statistics.SADAR_in_use.len * SADAR_PRICE * REQ_POINTS_WEIGHT
+	actualPoints += GLOB.monitor_statistics.B17_in_use.len * B17_PRICE * REQ_POINTS_WEIGHT
+	actualPoints += GLOB.monitor_statistics.B18_in_use.len * B18_PRICE * REQ_POINTS_WEIGHT
+	actualPoints += SSpoints.supply_points * REQ_POINTS_WEIGHT
+	actualPoints += GLOB.monitor_statistics.OB_available * OB_AVAILABLE_WEIGHT
+	return actualPoints
```

Comment:
Why not use . here?

Reply:
Does it offers anything? actualPoints is a named variable, it's more readable